### PR TITLE
Lesson03

### PR DESCRIPTION
**Pull request by volnsav**

### CODE REVIEW

File: 03-bash/task1.sh [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	echo "-h, --help - number of resulting file"
+	echo "-n         - number of files with hardware info"
+	echo "file       - file name with obtained hardware info"
+	echo "For example, ./task1.sh -n 9 ./test/info/info.txt"
+}
+
+# check and create work directory
+chk_work_dir(){
+	if ! [[ -d "$1" ]]; then
+	mkdir -p $1
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: indentation is missed

Comment (yekovalyov) [reply 03-bash/task1.sh]: option '-p ' allows to create dir even it already exists. In this context it isn't required.

File: 03-bash/task1.sh [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+		if ! [ -d "$1" ]; then
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: [[ is bash extension for [ operator. That is the reason for using different operators for same functionality?

File: 03-bash/task1.sh [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+   		return 1
+	fi
+}
+
+parse_number(){
+	local m_number=$2
+	if chk_is_number $1 -eq 0 ; then
+		eval $m_number=$1
+	fi
+	}
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: The indentations are different

File: 03-bash/task1.sh [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: trailing multiple empty lines in the end of file

File: 03-bash/task1.sh [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+# check and create work directory
+chk_work_dir(){
+	if ! [[ -d "$1" ]]; then
+	mkdir -p $1
+		if ! [ -d "$1" ]; then
+			>&2 echo "Error, working directory $1 was not created. Exit  with error."
+			exit 1
+		fi
+	else
+		echo "Directoy $1 is exists."
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: Typing error 'Directory'. Syntax error 'is exists'. Should be 'exits' (verb in Present Simple)

File: 03-bash/task1.sh [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+U_DIRNAME=""
+
+
+while [ "$1" != "" ]; do
+    case $1 in
+        -h |--help) 
+        help_usage
+		exit 0
+		;;
+		-n)
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: Indentation is incorrect

File: 03-bash/task1.sh (hunk 1 above)
Comment (volnsav) [reply 03-bash/task1.sh]: no, option '-p' allows to create directory tree.

File: 03-bash/task1.sh [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+	}
+	
+parse_file_name(){
+	local m_filename=$2
+	local m_dirname=$3
+	echo "fffff>>>  $m_filename"
+	echo "ddddd>>>> $m_dirname"
+	echo "11111>>>> $1"
+	if ! [[ -z "$1" ]] ; then
+		eval $m_filename=$(basename $1)
```
[the remark this answers is not in the document]
Comment (volnsav) [reply 03-bash/task1.sh]: yes, it save 'dirname' and 'basename' in variables.

File: 03-bash/task1.sh (hunk 2 above)
Comment (volnsav) [reply 03-bash/task1.sh]: ok, I'll  change all [[ to [

**Pull request closed without merging**